specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2023-02-01-preview/purviewcatalog.json

```diff
@@ -0,0 +1,8540 @@
+        "operationId": "Glossary_GetBulkDeleteOperationStatus",
+        "x-ms-examples": {
+          "Glossary_GetBulkDeleteOperationStatus": {
+            "$ref": "./examples/Glossary_GetBulkDeleteOperationStatusByOperationguid.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/operationGuid"
+          },
+          {
+            "$ref": "#/parameters/purviewAPIVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Return the status of bulk delete operation.",
+            "schema": {
+              "$ref": "#/definitions/BulkDeleteOperation"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/glossary/terms:delete": {
+      "post": {
+        "tags": [
+          "Glossary"
+        ],
+        "description": "Delete Glossary Terms",
+        "operationId": "Glossary_BulkDeleteTermsByGuids",
+        "x-ms-examples": {
+          "Glossary_BulkDeleteTermsByGuids": {
+            "$ref": "./examples/Glossary_GetBulkDeleteOperationStatus.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/forceDeleteEntityAssignment"
+          },
+          {
+            "$ref": "#/parameters/forceDeleteChild"
+          },
+          {
+            "$ref": "#/parameters/purviewAPIVersion"
+          },
+          {
+            "in": "body",
+            "name": "termGuids",
+            "description": "An array of term guids to be deleted.",
+            "required": true,
+            "schema": {
+              "type": "array",
+              "description": "An array of term guids to be deleted.",
+              "items": {
+                "$ref": "#/definitions/TermGuid"
+              }
+            }
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted. A job to delete glossary terms has been accepted.",
+            "headers": {
+              "Operation-Location": {
+                "description": "Response header for long running operation",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        }
+      }
+    },
+    "/glossary/terms": {
+      "put": {
+        "tags": [
+          "Glossary"
+        ],
+        "description": "Update terms in bulk.",
```

Comment:
Does this PUT completely replace all terms, or just add to the set of terms? If it is the latter, then the method should be POST rather than PUT.

Reply:
Yes. It will replace all terms. So we use PUT instead of POST.

Reply:
It looks like the Apache API implements this as a POST and not PUT:

https://atlas.apache.org/api/v2/ui/index.html#/

Why the change to use PUT here?

Reply:
Atlas has a create term API using POST and an update term API using PUT.
This API is bulk update term API, so we use PUT instead of POST.